### ISSUE TITLE
make test data mandatory?

### ISSUE THREAD

**Issue by jeromedockes** [probabl-ai/skore#2627]
### What would you like to say?

ATM a report can be created without X_test and y_test. I get that training data is optional when we pass a fitted estimator, but maybe we should force a test set to be provided?

**Comment by jeromedockes** [comment]
somewhat related to #2435

**Comment by glemaitre** [comment]
I agree that we should be more restrictive. I think that in the past we thought that `X_y` could be used and not force the test data but it does not make sense anymore.
